library/datasort.c

```diff
@@ -560,6 +561,70 @@ static int datasort_split_iterator_free(struct eblob_iterate_control *ictl __att
 	return 0;
 }
 
+/*
+ * Create a view over a blob and insert it in dcfg->unsorted_list
+ * Assume that blob has already sorted
+ */
+static int datasort_add_view_chunk(struct datasort_cfg *dcfg, struct eblob_base_ctl *bctl) {
+	assert(dcfg);
+	assert(bctl);
+
+	int err = 0;
+
+	EBLOB_WARNX(dcfg->log, EBLOB_LOG_INFO, "defrag: datasort_add_view_chunk: start, name: %s",
+			bctl->name);
+
+	struct datasort_chunk *chunk = calloc(1, sizeof(*chunk));
+	if (chunk == NULL) {
+		EBLOB_WARNC(dcfg->log, EBLOB_LOG_ERROR, errno,
+			    "defrag: add_view_chunk: can't allocate memory for datasort_chunk");
+		return -errno;
+	}
+
+	eblob_base_wait_locked(bctl);
+
+	struct eblob_file_ctl *index_ctl = &bctl->index_ctl;
+	assert(index_ctl->fd > 0);
+
+	chunk->fd = bctl->data_ctl.fd;
```

Comment:
This `fd` could be double closed by chunk destroy and base destroy, couldn't it?

Reply:
Solved: there is no close during destroy for view chunks.